packages/logger/src/formatter/LogFormatter.ts

```diff
@@ -19,8 +19,14 @@ abstract class LogFormatter implements LogFormatterInterface {
    */
   protected envVarsService?: EnvironmentVariablesService;
 
+  /**
+   * An array of keys that defines the order of the log record.
+   */
+  protected logRecordOrder?: LogRecordOrder;
+
   public constructor(options?: LogFormatterOptions) {
     this.envVarsService = options?.envVarsService;
+    this.logRecordOrder = options?.logRecordOrder;
```

Comment:
I've been thinking about this a bit (also why the review took a bit longer), and I think this property - and feature - should be only available in the `PowertoolsLogFormatter` class.

My reasoning is that if customers are creating their own log formatter class by extending `LogFormatter`, then they already have full control over the order of the keys - so having this available doesn't add much and instead pollutes the API.

I think we should position this feature as a way to lightly extend the default log formatter without having to create your own. This also matches what the Python implementation of Powertools does.

To sum up, I would move the `logRecordOrder` property in the `PowertoolsLogFormatter` and use it only there. Additionally, I would also extend the `ConstructorOptions` type to make `logRecordOrder` & `logFormatter` mutually exclusive, [similar to what we do for this other use case](https://github.com/aws-powertools/powertools-lambda-typescript/blob/23a33dc52db9d2bb8cfa8ff043abcc1020ff958e/packages/logger/src/types/Logger.ts#L131).